app/interactors/images/destroy.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class Images::Destroy
+  include Interactor
+  delegate :params, :user, to: :context
+
+  def initialize(params:, user:)
+    super
+  end
+
+  def call
+    Edition.find_and_lock_current(document: params[:document]) do |edition|
+      image_revision = edition.image_revisions.find_by!(image_id: params[:image_id])
+      updater = Versioning::RevisionUpdater.new(edition.revision, user)
```

Comment:
Yup, the create one was a lot closer to the feature test style I used at first. Create also has the advantage that not many variables are shared around so it can be quite simple. Whereas here we're passing the updater back.

This one could be along the lines of below, if we're not too concerned about context setting side-effects in methods.

```
def call
  Edition.find_and_lock_current(document: params[:document]) do |edition|
     remove_image(edition, params[:image_id])
     create_timeline_entry(edition)
     try_create_preview(edition)
  end
  
  def remove_image(edition, image_id)
    context.image_revision = edition.image_revisions.find_by!(image_id: params[:image_id])
    context.updater = Versioning::RevisionUpdater.new(edition.revision, user)

    updater.remove_image(image_revision)
    edition.assign_revision(updater.next_revision, user).save!
  end
end
```

Reply:
I found there are pros and cons to this: within the same file, it's quite easy to understand where all the assignment are coming from, but accessing the values becomes more verbose e.g. you're missing context in `updater.remove_image` in the above example. This is something I was toying with getting around by using something like `delegate_missing_to`, but I reckon what you have in your Create example is the best compromise, as it's really clear where the context is getting set. So, I vote for Create :-).